lib/src/client.dart

```diff
@@ -1158,12 +1161,222 @@ class Client extends MatrixApi {
     _archivedRooms.add(ArchivedRoom(room: archivedRoom, timeline: timeline));
   }
 
+  GetVersionsResponse? _versionsCache;
+
+  Future<bool> authenticatedMediaSupported() async {
+    _versionsCache ??= await getVersions();
+    return _versionsCache?.versions.any(
+          (v) => isVersionGreaterThanOrEqualTo(v, 'v1.11'),
+        ) ??
+        false;
+  }
```

Comment:
I know I am late, but imo it makes more sense to just try the authenticated media endpoints and see, if they fail. There are servers that claim 1.11 support for example, where the proxies are misconfigured and the endpoints just fail. We shouldn't fail to download media in those cases, but fall back to trying the old endpoints imo.

Reply:
moved to https://github.com/famedly/matrix-dart-sdk/pull/1903